dbt/adapters/databricks/python_models/python_submissions.py

```diff
@@ -138,189 +224,318 @@ def _submit_job(self, path: str, cluster_spec: dict) -> str:
+        config = parsed_model.config
+        if config.python_job_config:
+            cluster_settings.update(config.python_job_config.additional_task_settings)
+            workflow_spec = config.python_job_config.dict()
+            workflow_spec["name"] = PythonWorkflowConfigCompiler.workflow_name(parsed_model)
+            existing_job_id = config.python_job_config.existing_job_id
+            post_hook_tasks = config.python_job_config.post_hook_tasks
+            return PythonWorkflowConfigCompiler(
+                cluster_settings, workflow_spec, existing_job_id, post_hook_tasks
             )
+        else:
+            return PythonWorkflowConfigCompiler(cluster_settings, {}, "", [])
+
+    @staticmethod
+    def workflow_name(parsed_model: ParsedPythonModel) -> str:
+        name: Optional[str] = None
+        if parsed_model.config.python_job_config:
+            name = parsed_model.config.python_job_config.name
+        return (
+            name or f"dbt__{parsed_model.catalog}-{parsed_model.schema_}-{parsed_model.identifier}"
+        )
 
-    def submit(self, compiled_code: str) -> None:
-        workflow_spec = self._build_job_spec()
-        self._submit_through_workflow(compiled_code, workflow_spec)
+    @staticmethod
+    def cluster_settings(parsed_model: ParsedPythonModel) -> Dict[str, Any]:
+        config = parsed_model.config
+        job_cluster_config = config.job_cluster_config
 
-    def _build_job_spec(self) -> Dict[str, Any]:
-        workflow_spec = dict(self.workflow_spec)
-        workflow_spec["name"] = self.workflow_spec.get("name", self.default_job_name)
+        cluster_settings: Dict[str, Any] = {}
+        if job_cluster_config:
+            cluster_settings["new_cluster"] = job_cluster_config
+        elif config.cluster_id:
+            cluster_settings["existing_cluster_id"] = config.cluster_id
 
-        # Undefined cluster settings defaults to serverless in the Databricks API
-        cluster_settings = {}
-        if self.cluster_spec:
-            cluster_settings["new_cluster"] = self.cluster_spec
-        elif "existing_cluster_id" in self.workflow_spec:
-            cluster_settings["existing_cluster_id"] = self.workflow_spec["existing_cluster_id"]
+        return cluster_settings
 
+    def compile(self, path: str) -> Tuple[Dict[str, Any], str]:
         notebook_task = {
             "task_key": "inner_notebook",
             "notebook_task": {
-                "notebook_path": self.notebook_path,
+                "notebook_path": path,
                 "source": "WORKSPACE",
             },
         }
-        notebook_task.update(cluster_settings)
-        notebook_task.update(self.additional_task_settings)
+        notebook_task.update(self.task_settings)
+
+        self.workflow_spec["tasks"] = [notebook_task] + self.post_hook_tasks
+        return self.workflow_spec, self.existing_job_id
+
+
+class PythonWorkflowCreater:
```

Comment:
PythonWorkflowCreator?

Reply:
I was very much on the fence about the proper spelling, happy to change :)